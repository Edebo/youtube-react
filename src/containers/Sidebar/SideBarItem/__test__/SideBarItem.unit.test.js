import React from "react";
import { shallow } from "enzyme";
import { SideBarItem } from "../SideBarItem";

describe("testing side bar item", () => {
  //   test("rendering side bar item withour props", () => {
  //     const wrapper = shallow(<SideBarItem />);

  //     expect(wrapper).toMatchSnapShot();
  //   });
  test("renders empty SideBarItem", () => {
    const wrapper = shallow(<SideBarItem />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders highlighted SideBarItem that navigates to /feed/trending", () => {
    const wrapper = shallow(
      <SideBarItem highlighted icon='fire' label='Trending' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("render non-highlighted SideBarItem that navigates to /feed/trending", () => {
    const wrapper = shallow(<SideBarItem icon='fire' label='Trending' />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Renders highlighted SideBarItem with no navigation", () => {
    const wrapper = shallow(
      <SideBarItem highlighted icon='fire' label='Trending' />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
