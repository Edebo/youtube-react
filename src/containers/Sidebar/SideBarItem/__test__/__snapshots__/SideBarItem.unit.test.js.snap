// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing side bar item Renders highlighted SideBarItem with no navigation 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
       
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`testing side bar item render non-highlighted SideBarItem that navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
       
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`testing side bar item renders empty SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
       
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`testing side bar item renders highlighted SideBarItem that navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
       
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
